### PR TITLE
sipify gui part 1

### PR DESCRIPTION
**Pull request by 3nids**
## Description
Include a few sentences describing the overall goals for this PR (pull request). If applicable also add screenshots.

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [ ] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [ ] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [ ] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and containt sufficient information in the commit message to be documented
- [ ] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] New unit tests have been added for core changes
- [ ] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

### CODE REVIEW

File: python/gui/qgsattributeform.sip [hunk 1]
```diff
@@ -1,210 +1,227 @@
...
-      FilterAnd, /*!< Filter should be combined using "AND" */
-      FilterOr, /*!< Filter should be combined using "OR" */
+      ReplaceFilter,
+      FilterAnd,
+      FilterOr,
     };
 
-    explicit QgsAttributeForm( QgsVectorLayer *vl, const QgsFeature &feature = QgsFeature(), const QgsAttributeEditorContext &context = QgsAttributeEditorContext(), QWidget *parent /TransferThis/ = 0 );
+    explicit QgsAttributeForm( QgsVectorLayer *vl, const QgsFeature &feature = QgsFeature(),
+                               const QgsAttributeEditorContext &context = QgsAttributeEditorContext(), QWidget *parent = 0 );
```
Comment (nyalldawson) [inline python/gui/qgsattributeform.sip]: missing TransferThis

File: python/gui/qgsattributeform.sip [hunk 2]
```diff
@@ -1,210 +1,227 @@
...
+    void beforeSave( bool &ok );
+%Docstring
+ Will be emitted before the feature is saved. Use this signal to perform sanity checks.
+ You can set the parameter ok to false to notify the form that you don't want it to be saved.
+ If you want the form to be saved, leave the parameter untouched.
+
+ \param ok  Set this parameter to false if you don't want the form to be saved
+.. note::
+
+   not available  in Python bindings
```
Comment (nyalldawson) [inline python/gui/qgsattributeform.sip]: This comment should be removed?

File: python/gui/qgsattributeformeditorwidget.sip [hunk 3]
```diff
@@ -1,130 +1,170 @@
...
     ~QgsAttributeFormEditorWidget();
 
-    /** Creates the search widget wrappers for the widget used when the form is in
-     * search mode.
-     * @param widgetId id of the widget type to create a search wrapper for
-     * @param fieldIdx index of field associated with widget
-     * @param config configuration which should be used for the widget creation
-     * @param context editor context (not available in python bindings)
-     */
-    void createSearchWidgetWrappers( const QString& widgetId, int fieldIdx,
```
Comment (nyalldawson) [inline python/gui/qgsattributeformeditorwidget.sip]: This has disappeared - was that intentional?

File: python/gui/qgsbusyindicatordialog.sip [hunk 4]
```diff
@@ -1,23 +1,48 @@
...
-    /** Constructor
-     * Modal busy indicator dialog with no buttons.
-     * @param message Text to show above busy progress indicator.
-     * @param parent parent object (owner)
-     * @param fl widget flags
-     */
-    QgsBusyIndicatorDialog( const QString &message = "", QWidget *parent /TransferThis/ = 0, const Qt::WindowFlags &fl = QgsGuiUtils::ModalDialogFlags );
-    ~QgsBusyIndicatorDialog();
+
+    QgsBusyIndicatorDialog( const QString &message = "", QWidget *parent = 0, Qt::WindowFlags fl = QgsGuiUtils::ModalDialogFlags );
```
Comment (nyalldawson) [inline python/gui/qgsbusyindicatordialog.sip]: Missing TransferThis

File: python/gui/qgsattributeform.sip (hunk 2 above)
Comment (3nids) [inline python/gui/qgsattributeform.sip]: adding SIP_SKIP, method was commented out before

File: python/gui/qgsattributeformeditorwidget.sip (hunk 3 above)
Comment (3nids) [inline python/gui/qgsattributeformeditorwidget.sip]: nope, it was the argument which needed to be removed

**Pull request merged**